lib/dal/src/workspace_snapshot.rs

```diff
@@ -745,6 +767,89 @@ impl WorkspaceSnapshot {
         })
     }
 
+    /// Attempts to migrate a snapshot graph to the latest version. Should only
```

Comment:
Clever! So we don't actually need a version mapping explicitly then? What if we need to make another change to EdgeWeight? Or anything else about the WorkspaceSnapshotGraph needs to change?